service-registry-client/src/main/java/no/difi/meldingsutveksling/serviceregistry/ServiceRegistryClient.java

```diff
@@ -0,0 +1,87 @@
+package no.difi.meldingsutveksling.serviceregistry;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.nimbusds.jose.proc.BadJWSException;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import no.difi.meldingsutveksling.serviceregistry.client.RestClient;
+import no.difi.meldingsutveksling.serviceregistry.externalmodel.IdentifierResource;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestClientException;
+
+import java.io.IOException;
+import java.util.List;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class ServiceRegistryClient {
+
+    public static final String CACHE_LOAD_IDENTIFIER_RESOURCE = "loadIdentifierResource";
+    public static final String CACHE_GET_SAS_KEY = "getSasKey";
+
+    private final RestClient client;
+    private final SasKeyRepository sasKeyRepository;
+    private final ObjectMapper objectMapper;
+
+    @Cacheable(CACHE_LOAD_IDENTIFIER_RESOURCE)
+    public IdentifierResource loadIdentifierResource(SRParameter parameter) throws ServiceRegistryLookupException {
+        String identifierResourceString = getIdentifierResourceString(parameter);
+
+        try {
+            return objectMapper.readValue(identifierResourceString, IdentifierResource.class);
+        } catch (IOException e) {
+            throw new ServiceRegistryLookupException(
+                    String.format("Parsing response as a IdentifierResource JSON object failed. Content is: %s", identifierResourceString)
+                    , e);
+        }
+    }
+
+    private String getIdentifierResourceString(SRParameter parameter) throws ServiceRegistryLookupException {
+        try {
+            return client.getResource("identifier/" + parameter.getIdentifier(), parameter.getQuery());
+        } catch (HttpClientErrorException httpException) {
+            byte[] errorBody = httpException.getResponseBodyAsByteArray();
+            try {
+                ErrorResponse error = objectMapper.readValue(errorBody, ErrorResponse.class);
+                throw new ServiceRegistryLookupException(String.format("Caught exception when looking up service record with identifier %s, http status %s (%s): %s",
+                        parameter.getIdentifier(), httpException.getStatusCode(), httpException.getStatusText(), error.getErrorDescription()), httpException);
+            } catch (IOException e) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1166)